client/state/comments/middleware.js

```diff
@@ -0,0 +1,27 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { COMMENTS_CHANGE_STATUS } from 'state/action-types';
+import { getSiteComment } from 'state/selectors';
+
+const handler = ( dispatch, action, getState ) => {
+	switch ( action.type ) {
+		case COMMENTS_CHANGE_STATUS:
+			const siteComment = getSiteComment( getState(), action.siteId, action.commentId );
+			const previousStatus = siteComment && siteComment.status;
+			return {
+				...action,
+				meta: Object.assign( {}, action.meta, { comment: { previousStatus } } ),
```

Comment:
Note that when we modifying actions in a data-layer handler, this change isn't passed down the chain. We recalculate status here so the counts reducer also has access to `previousStatus` without forcing us to update the signature of the `COMMENTS_CHANGE_STATUS` action.